arangod/Scheduler/Socket.h

```diff
@@ -85,10 +86,10 @@ class Socket {
   }
 
   void post(std::function<void()> handler) {
-    _scheduler->post(*_strand, handler);
+    _context.post(handler);
   }
 
-  bool runningInThisThread() { return _strand->running_in_this_thread(); }
+  bool runningInThisThread() { return true; }
```

Comment:
This rings a (small) alarm bell. We need to check that we do not actually run any request in the io thread and places where this is called might hint us as to where this is happening.

Reply:
Agree. I will create a `runningInThisThread` functionality for the `IoContext`. In addition to that there was only one place where it was not used in a assertion. However, due to other conditions evaluating always true, a potentially problematic code path was never taken.